olm/README.md

```diff
@@ -9,73 +9,205 @@ WARNING: Please make sure to use the precise `v0.10.0` version of the `operator-
 
-```shell
-$ export QUAY_ECLIPSE_CHE_USERNAME=${username} && \
-export QUAY_ECLIPSE_CHE_PASSWORD=${password} && \
-export APPLICATION_REGISTRY=${application_registry_namespace} && \
-./push-olm-files-to-quay.sh
+To test nightly che-operator you have to use Olm CatalogSource(index) image. 
+CatalogSource image stores in the internal database information about Olm bundles with different versions of the Eclipse Che.
+For nightly channel (dependent on platform) Eclipse Che provides two CatalogSource images:
+ 
+ - `quay.io/eclipse/eclipse-che-kubernetes-opm-catalog:preview` for kubernetes platform;
+ - `quay.io/eclipse/eclipse-che-openshift-opm-catalog:preview` for openshift platform;
+
+For each new nightly version Eclipse Che provides nightly bundle image with name pattern:
+
+`quay.io/eclipse/eclipse-che-${platform}-opm-bundles:${cheVersion}-${incrementVersion}.nightly`
+
+For example:
+
+```
+quay.io/eclipse/eclipse-che-kubernetes-opm-bundles:7.18.0-1.nightly
+...
+quay.io/eclipse/eclipse-che-kubernetes-opm-bundles:7.19.0-5.nightly
+...
 ```
 
-Go to the quay.io and use ui(tab Settings) to make your application public.
-Start minikube(or CRC) and after that launch test script in the olm folder:
+To test the latest "nightly" bundle use `olm/TestCatalogSource.sh` script:
 
-```shell
-$ export APPLICATION_REGISTRY=${application_registry_namespace} && ./testCSV.sh ${platform} ${package_version} ${optional-namespace}
+```bash
+$ ./testCatalogSource.sh ${platform} "nightly" ${namespace} "catalog"
 ```
 
-Where are:
- - `platform` - 'openshift' or 'kubernetes'
- - `package_version` - your generated che-operator package version(for example: `7.8.0` or `9.9.9-nightly.1562083645`)
- - `optional-namespace` - kubernetes namespace to deploy che-operator. Optional parameter, by default operator will be deployed to the namespace `eclipse-che-preview-test`
+To test migration Che from previous nightly version to the latest you can use `olm/testUpdate.sh` script:
 
-To test che-operator with OLM files without push to a related Quay.io application, we can build a required docker image of a dedicated catalog,
-in order to install directly through a CatalogSource. To test this options start minikube and after that launch
-test script in the olm folder:
+```bash
+$ ./testUpdate.sh ${platform} "nightly" ${namespace}
+```
+
+See more information about test arguments in the chapter: [Test arguments](#test-script-arguments)
+
+### 7. Build custom nightly bundle images
+
+For test purpose you can build your own "nightly" CatalogSource and bundle images
+with your latest development changes and use it in the test scripts.
+To build these images you can use script `olm/buildFirstBundle.sh`:
+
+```bash
+$ export IMAGE_REGISTRY_USER_NAME=${userName} && \
+  export IMAGE_REGISTRY_PASSWORD=${password} && \
+  export IMAGE_REGISTRY_HOST=${imageRegistryHost} && \
+  ./buildFirstBundle.sh ${platform} ${optional-from-index-image}
+```
```

Comment:
I would rename `buildFirstBundle.sh` to `buildAndPushInitialBundle.sh`

Reply:
done